### ISSUE TITLE
`output.library.root` should accept an array

### ISSUE THREAD

**Issue by aj-r** [webpack/webpack#7176]
**Do you want to request a *feature* or report a *bug*?** 

Bug?

**What is the current behavior?**

In `webpack.conf.js`:

```javascript
export = {
    output: {
        libraryTarget: "umd",
        library: {
            amd: "my-library",
            commonjs: "my-library",
            root: ["My", "Library"],
        }
    }
};
```

Result:

```
Invalid configuration object. Webpack has been initialised using a configuration object that does not match the API schema.
 - configuration.output.library should be one of these:
   string | [string] | object { root?, amd?, commonjs? }
   -> If set, export the bundle as library. `output.library` is the name.
   Details:
    * configuration.output.library should be a string.
    * configuration.output.library should be an array:
      [string]
    * configuration.output.library.root should be a string.
      -> Name of the property exposed globally by a UMD library
```

**If the current behavior is a bug, please provide the steps to reproduce.**

Create the configuration above and run `webpack`.

Note that I can pass an array directly to the library as shown below. In that case, the correct object is added to `root`, but `amd` and `commonjs` are registered with the wrong name.

```javascript
export = {
    output: {
        libraryTarget: "umd",
        library: ["My", "Library"],
    }
};
```

So why can I pass an array to `library`, but not to `library.root`?

**What is the expected behavior?**

No error.

**If this is a feature request, what is motivation or use case for changing the behavior?**

**Please mention other relevant information such as the browser version, Node.js version, webpack version, and Operating System.**

- webpack: 4.6.0
- node: 8.9.4
- OS: Windows 8.1

**Comment by ooflorent** [comment]
Using an array is not something we want to encourage.
`library` can be an array for compatibility reason.

**Comment by aj-r** [comment]
Why is it not encouraged? I need my library to exist on `root["My"]["Library"]` for compatibility reasons.

**Comment by sokra** [comment]
Send a PR

**Comment by byzyk** [comment]
@aj-r tried to fix that in #7184. Would you be able to check if it works for your case?

**Comment by aj-r** [comment]
@byzyk looks good from my perspective :)
